sound/soc/sof/intel/hda.h

```diff
@@ -856,4 +857,7 @@ irqreturn_t hda_dsp_ipc4_irq_thread(int irq, void *context);
 int hda_dsp_ipc4_send_msg(struct snd_sof_dev *sdev, struct snd_sof_ipc_msg *msg);
 extern struct sdw_intel_ops sdw_callback;
 
+struct sof_ipc4_fw_library;
```

Comment:
?  what's this ?

Reply:
forward declaration so we don't need to include the whole `ipc4-priv.h` to `hda.h`.

Reply:
advance usage